adapter/prometheus2/prometheus.go

```diff
@@ -275,3 +325,13 @@ func promLabels(l map[string]interface{}) prometheus.Labels {
 	}
 	return labels
 }
+
+func computeSha(m *config.Params_MetricInfo) [sha1.Size]byte {
```

Comment:
this is one way to do it.

curious: did we consider building a Metric and then using the prometheus code (`metric.Fingerprint()` or even `metric.Equal()` ), etc. to do the checking instead?

It might be good to ensure the generated metrics do not collide, rather than just the templates for the metrics (and would reduce differences in models and hashing, etc.).

Reply:
This is using the exact information needed to construct a collector. `{kind, name, desc, labels}`. Are you proposing we use the collector itself or actually construct a fake value ?

Can metrics / collectors defined by 2 different configs collide with each other? they **will** have different names, correct?